terraform/examples/daos_cluster/variables.tf

```diff
@@ -44,81 +37,117 @@ variable "server_labels" {
 }
 
-variable "server_machine_type" {
-  description = "GCP machine type. e.g. e2-medium"
-  default     = null
+variable "server_template_name" {
+  description = "MIG template name"
+  default     = "daos-server"
   type        = string
 }
 
-variable "server_os_family" {
-  description = "OS GCP image family"
-  default     = null
+variable "server_mig_name" {
+  description = "MIG name "
+  default     = "daos-server"
   type        = string
 }
 
-variable "server_os_project" {
-  description = "OS GCP image project name"
-  default     = null
+variable "server_machine_type" {
+  description = "GCP machine type. ie. e2-medium"
+  default     = "n2-custom-36-215040"
   type        = string
 }
 
-variable "server_os_disk_type" {
-  description = "OS disk type e.g. pd-ssd, pd-standard"
-  default     = "pd-ssd"
+variable "server_instance_base_name" {
+  description = "MIG instance base names to use"
+  default     = "daos-server"
   type        = string
 }
 
-variable "server_os_disk_size_gb" {
-  description = "OS disk size in GB"
-  default     = 20
+variable "server_number_of_instances" {
+  description = "Number of daos servers to bring up"
+  default     = 4
   type        = number
 }
 
+variable "server_daos_disk_type" {
+  #TODO: At some point we will support more than local-ssd with NVME
+  # interface.  This variable will be useful then. For now its just this.
+  description = "Daos disk type to use. For now only suported one is local-ssd"
+  default     = "local-ssd"
+  type        = string
+}
+
 variable "server_daos_disk_count" {
   description = "Number of local ssd's to use"
-  default     = null
+  default     = 16
   type        = number
 }
 
-variable "server_daos_crt_timeout" {
-  description = "crt_timeout"
-  default     = null
-  type        = number
+variable "server_service_account" {
+  description = "Service account to attach to the instance. See https://www.terraform.io/docs/providers/google/r/compute_instance_template.html#service_account."
+  type = object({
+    email  = string,
+    scopes = set(string)
+  })
+  default = {
+    email = null
+    scopes = ["https://www.googleapis.com/auth/devstorage.read_only",
+      "https://www.googleapis.com/auth/logging.write",
+      "https://www.googleapis.com/auth/monitoring.write",
+      "https://www.googleapis.com/auth/servicecontrol",
+      "https://www.googleapis.com/auth/service.management.readonly",
+    "https://www.googleapis.com/auth/trace.append"]
+  }
+}
+
+variable "server_preemptible" {
+  description = "If preemptible instances"
+  default     = false
+  type        = string
+}
+
+variable "server_pools" {
+  description = "If provided, this module will generate a script to create a list of pools. pool attributes have to be specified in a format acceptable by [dmg](https://docs.daos.io/v2.0/admin/pool_operations/) and daos."
+  default     = []
+  type = list(object({
+    pool_name  = string
+    pool_size  = string
+    containers = list(string)
```

Comment:
Containers have a broad config set: https://docs.daos.io/v2.0/user/container/#container-properties .  As we talked about today this is how ACLs would be added.   Replication and EC also are set at pool creation time.   I think the improvement  would be move from a basic string type to a full DAOS_Container type.   Fully agreed not to add all there here in this PR.